Assignment 5.Rmd

```diff
@@ -41,40 +48,92 @@ We want to see if we can build a decision tree to help teachers decide which stu
 ```{r}
-D1$advice <- 
+
+for (i in 1:nrow(D1)){
+  if(as.numeric(D1$score)[i]>=.9){
+    D1$advice[i]<- "No action"
+  }else if(as.numeric(D1$score)[i]<.9){
+    D1$advice[i]<- "Teacher should monitor"
+  }else if(as.numeric(D1$score)[i]<.6){
+    D1$advice[i]<- "Teacher should intervene"
+  }
+}
+
+D1$advice<-as.character(D1$advice)
+
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+
+score_tree <- rpart(advice ~ prior_prob_count + prior_percent_correct + hints, method="class", data=D1) #Notice the standard R notion for a formula X ~ Y 
+
 ```
 
 #Plot tree
 ```{r}
+printcp(score_tree)
+rpart.plot(score_tree)
+post(c.tree, file = "tree2.ps", title = "Session Completion Action: 1 - Ask teacher, 2 - Start new session, 3 - Give up")
 
 ```
 
-Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+Please interpret the tree, which two behaviors do you think the teacher should most closely pay attention to?
+
+Based on the generated plot, it seems as though the most powerful predictors of whether or not a student will receive feedback of some sort is whether they hace received such hunts in the past. If a student has not received many hints, and falls within a narrow stretch of students that either had mid to high mid scores and completed a certain amount of problems would the program suggest that we give them some form of advice. 
 
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("~/tmp/assignment5/intelligent_tutor.csv", stringsAsFactors=FALSE)
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+D2$prediction <- predict(score_tree, D2, type = "class")
 
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+for (i in 1:nrow(D2)){
+  if(as.numeric(D2$score)[i]>=.9){
+    D2$advice[i]<- "No action"
+  }else if(as.numeric(D2$score)[i]<.9){
+    D2$advice[i]<- "Teacher should monitor"
+  }else if(as.numeric(D2$score)[i]<.6){
+    D2$advice[i]<- "Teacher should intervene"
+  }
+}
+
+for (i in 1:nrow(D2)){
+  if(D2$prediction[i]==D2$advice[i]){
+    D2$match[i]<-"Y"
+  }else{
+    D2$match[i]<-"N"
+  }
+}
+
+library(dplyr)
+D3<-D2 %>%
+  group_by(match)%>%
+  count(match)%>%
+  summarise(match/nrow(D3$match))
+
+
+
+```
+
```

Comment:
So does that mean that the model overfits the data?

Reply:
I don't think so- overfitting is a situation where you fit your training data too closely to account for the "noise of reality" that is present in our testing data. If I were to do this all over again, I would look at a ROC curve and use the probability mode of rpart to modulate for when our tree plot funneled a user into a specific box.